operator/internal/controller/paladinregistration_controller.go

```diff
@@ -137,6 +137,9 @@ func (r *PaladinRegistrationReconciler) Reconcile(ctx context.Context, req ctrl.
 			return ctrl.Result{}, err
 		} else if regTx.statusChanged {
 			reg.Status.PublishTxs[transportName] = transportPublishStatus
+			if transportPublishStatus.TransactionStatus == corev1alpha1.TransactionStatusSuccess {
+				publishCount++
```

Comment:
I notice there's still a `publishCount++` in the loop below, which maybe is in practice unreachable code?
... probably means this should be a `switch` statement, and all paths should return

Reply:
ah, no... surely we need to go round the loop for each one - so the problem is that `publishCount++` was been incremented too late before...

Reply:
```go
// This line need to changed into `statusUpdated = true` and then AFTER the loop, we need to return `updateStatusAndRequeue`
return r.updateStatusAndRequeue(ctx, &reg, publishCount)
```